In [1]:
#graphlab.get_dependencies()

In [2]:
print "---------------------"
import graphlab

---------------------


In [3]:
song_data = graphlab.SFrame('../data/song_data.gl/')
print song_data.head(3)

[INFO] graphlab.cython.cy_server: GraphLab Create v2.1 started. Logging: C:\Users\bmt\AppData\Local\Temp\graphlab_server_1480218185.log.0


This non-commercial license of GraphLab Create for academic use is assigned to bmatthewtaylor@gmail.com and will expire on October 27, 2017.
+-------------------------------+--------------------+--------------+
|            user_id            |      song_id       | listen_count |
+-------------------------------+--------------------+--------------+
| b80344d063b5ccb3212f76538f... | SOAKIMP12A8C130995 |      1       |
| b80344d063b5ccb3212f76538f... | SOBBMDR12A8C13253B |      2       |
| b80344d063b5ccb3212f76538f... | SOBXHDL12A81C204C0 |      1       |
+-------------------------------+--------------------+--------------+
+-----------------+---------------+-------------------------------+
|      title      |     artist    |              song             |
+-----------------+---------------+-------------------------------+
|     The Cove    |  Jack Johnson |    The Cove - Jack Johnson    |
| Entre Dos Aguas | Paco De Lucia | Entre Dos Aguas - Paco De ... |
|     Stronger    |   Kanye W

In [4]:
kanye = song_data.filter_by('Kanye West', 'artist')
print type(kanye), kanye.shape
#print kanye.head(1)
print  "No of unique listeners = ", len(set(kanye['user_id']))
users = kanye['user_id']
print type(users), len(users)
import graphlab.aggregate as agg
user_summary = kanye.groupby(key_columns='user_id', operations={'user_id-sum': agg.COUNT()})
print  "No of unique users = ",user_summary.shape[0]


<class 'graphlab.data_structures.sframe.SFrame'> (3775, 6)
No of unique listeners =  2522
<class 'graphlab.data_structures.sarray.SArray'> 3775
No of unique users =  2522


In [5]:
fooFighters = song_data.filter_by('Foo Fighters', 'artist')
print "fooFighters.shape = ", fooFighters.shape
print "No of unique listeners = ", len(set(fooFighters['user_id']))
users = fooFighters['user_id']
print type(users), len(users)
import graphlab.aggregate as agg
user_summary = fooFighters.groupby(key_columns='user_id', operations={'user_id-sum': agg.COUNT()})
print  "No of unique users = ",user_summary.shape[0]


fooFighters.shape =  (3429, 6)
No of unique listeners =  2055
<class 'graphlab.data_structures.sarray.SArray'> 3429
No of unique users =  2055


In [6]:
taylorSwift = song_data.filter_by('Taylor Swift', 'artist')
print "taylorSwift.shape = ", taylorSwift.shape
print "No of unique listeners = ", len(set(taylorSwift['user_id']))
users = taylorSwift['user_id']
print type(users), len(users)
import graphlab.aggregate as agg
user_summary = taylorSwift.groupby(key_columns='user_id', operations={'user_id-sum': agg.COUNT()})
print  "No of unique users = ",user_summary.shape[0]


taylorSwift.shape =  (6227, 6)
No of unique listeners =  3246
<class 'graphlab.data_structures.sarray.SArray'> 6227
No of unique users =  3246


In [7]:
ladyGaga = song_data.filter_by('Lady GaGa', 'artist')
print "ladyGaga.shape = ", ladyGaga.shape
print "No of unique listeners = ", len(set(ladyGaga['user_id']))
users = ladyGaga['user_id']
print type(users), len(users)
import graphlab.aggregate as agg
user_summary = ladyGaga.groupby(key_columns='user_id', operations={'user_id-sum': agg.COUNT()})
print  "No of unique users = ",user_summary.shape[0]


ladyGaga.shape =  (4129, 6)
No of unique listeners =  2928
<class 'graphlab.data_structures.sarray.SArray'> 4129
No of unique users =  2928


subset dataframe by rows with 'artist' = artist name
count unique user_id - use set since easier than operating on list.
Kanye West    = 2522
Foo Fighters  = 2055
Taylor Swift  = 3246  [ANSWER = most popular = most unique users]
Lady GaGa     = 2928

In [8]:
print 'Taylor Swift\t\t', song_data.filter_by('Taylor Swift', 'artist')['listen_count'].sum()
print 'Kings of Leon\t\t', song_data.filter_by('Kings of Leon', 'artist')['listen_count'].sum()
print 'Coldplay\t\t', song_data.filter_by('Coldplay', 'artist')['listen_count'].sum()
print 'Lady GaGa\t\t', song_data.filter_by('Lady GaGa', 'artist')['listen_count'].sum()

Taylor Swift		19376
Kings of Leon		0
Coldplay		35362
Lady GaGa		12224


In [9]:
print 'William Tabbert\t\t\t', song_data.filter_by('William Tabbert', 'artist')['listen_count'].sum()
print 'Velvet Underground & Nico\t', song_data.filter_by('Velvet Underground & Nico', 'artist')['listen_count'].sum()
print 'Kanye West\t\t\t', song_data.filter_by('Kanye West', 'artist')['listen_count'].sum()
print 'The Cool Kids\t\t\t', song_data.filter_by('The Cool Kids', 'artist')['listen_count'].sum()

William Tabbert			14
Velvet Underground & Nico	80
Kanye West			9992
The Cool Kids			73


In [25]:
temp = song_data.filter_by('The Cool Kids', 'artist')
print type(temp), temp.shape

<class 'graphlab.data_structures.sframe.SFrame'> (39, 6)


In [26]:
print temp['listen_count'].head(10)
print temp['listen_count'].sum()

[1L, 4L, 1L, 4L, 1L, 1L, 1L, 7L, 1L, 3L]
73
